src/tdyregression.c

```diff
@@ -201,7 +201,7 @@ PetscErrorCode TDyRegressionOutput(TDy tdy, Vec U) {
   ierr = VecRestoreArray(U,&u_p); CHKERRQ(ierr);
   ierr = VecRestoreArray(U_pres,&pres_p); CHKERRQ(ierr);
 
-  if (tdy->mode == TH) {
+  if (tdy->options.mode == TH) {
```

Comment:
Also not pertinent to this step, but it would be nice to use callbacks instead of if statements, which will become `switch` statements if there are ever more than two modes.

Reply:
Can you give an example of what you mean by this?

Reply:
For example, the mode could have a `tdy->diagnostics(tdy, ...)` that computes/reports any diagnostics specific to that mode. This saves having a sequence of switch statements with possibly fragile carried dependencies.

Reply:
Ah, okay.  A "virtual table". Yes, that makes sense.